hikari/config.py

```diff
@@ -397,8 +397,72 @@ class that will **NOT** enforce SSL verification. This is then stored
 class CacheSettings:
     """Settings to control the cache."""
 
+    enable: bool = attr.ib(default=True)
+    """Whether to enable the cache.
+
+    If set to `False`, all the cache functionality will be disabled.
+
+    Defaults to `builtins.True`.
+    """
+
+    guilds: bool = attr.ib(default=True)
+    """Whether to enable the guilds cache.
+
+    Defaults to `builtins.True`.
+    """
+
+    members: bool = attr.ib(default=True)
+    """Whether to enable the members cache.
+
+    Defaults to `builtins.True`.
+    """
+
+    channels: bool = attr.ib(default=True)
```

Comment:
might want to call this guild_channels and specify that it's only guild channels to be consistent with the cache